next/src/components/atoms/AppLangSwitchers.tsx

```diff
@@ -1,78 +1,62 @@
 import cx from 'classnames'
 import { useRouter } from 'next/router'
-import { useTranslation } from 'next-i18next'
-import { useCallback } from 'react'
+import React, { useCallback } from 'react'
 
-import { ContentPageEntityFragment } from '@/src/services/graphql'
-import { WithAttributes } from '@/src/utils/isDefined'
+import { PageWrapperProps } from '@/src/components/pages/PageWrapper'
 import { getEquivalentRouteInTargetLocale } from '@/src/utils/localeRoutes'
 
-interface AppLangSwitchersProps {
-  contentPage?: WithAttributes<ContentPageEntityFragment>
-  showBothLanguages?: boolean
-}
+type AppLangSwitchersProps = { showBothLanguages?: boolean } & Pick<PageWrapperProps, 'page'>
 
-const AppLangSwitchers = ({ contentPage, showBothLanguages }: AppLangSwitchersProps) => {
+const AppLangSwitchers = ({ page, showBothLanguages }: AppLangSwitchersProps) => {
   const router = useRouter()
-  const { i18n } = useTranslation()
-  const currentLanguage = i18n.language
+
+  const { asPath, locale: currentLocale, locales } = router // Use router to access locale for better Next integration
```

Comment:
Please use `i18n.language` from `useTranslation()` as `locale` as we do in other projects